packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -165,10 +166,6 @@ export abstract class FluidDataStoreContext
 		return this._containerRuntime.logger;
 	}
 
-	public get deltaManager(): IDeltaManager<ISequencedDocumentMessage, IDocumentMessage> {
```

Comment:
What about ContainerRuntime.deltaManager getter? It feels like it needs to either be removed, or have a proxy object there - there should be no way to get to "real" think it seems.
If some scenario requires it, then maybe it should not be public, and appropriate comments / maybe even renames should be done to clearly articulate to callers why it's not virtualized same way as for data stores

Reply:
Summarizer will need access to real thing, but can we special case it somehow, and let the rest of the system (not just data stores) to work only with proxy?

Reply:
Good point. The getter should be removed.
Also:
- We should not be passing container runtime to data store context but pass either a smaller interface or the required properties directly.
- Delta manager should not be directly exposed from data store context / data store runtime. Only the required properties from it should be exposed directly. An example of how things can go wrong - IDetlaManager exposes methods like close(), dispose() and flush() method which if called can mess things up.

Reply:
Changed the getter to return the proxy. Added _deltaManager to ContainerRuntime that is used by that class. There are a few places in ContainerRuntime that needs to use the actual delta manager because they access readonly info. For consistency, I updated all usages in ContainerRuntime to use _deltaManager.